File: NeetCode/Arrays/Easy/392.IsSubsequence.py
#Given two strings s and t, return true if s is a subsequence of t, 
#or false otherwise.


def isSubsequence(s, t):
    if s == '':
        return True

    count = 0 

    for letter in t:
        if len(s) == count:
            return True
        if s[count] == letter:
            count += 1

    return True if len(s) == count else False

print(isSubsequence("abc", "ahbgdc"))
print(isSubsequence("axc", "ahbgdc"))

File: NeetCode/Swift/Arrays/Easy/392.IsSubsequence.swift
//Given two strings s and t, return true if s is a subsequence of t, 
//or false otherwise.


func isSubsequence(_ s: String, _ t: String) -> Bool {
    if s.isEmpty {
        return true
    }

    return t.reduce((s: Array(s), index: 0)) { (result, letter) in
        var (s, index) = result
        if index < s.count && s[index] == letter {
            index += 1
        }
        return (s, index)
    }.index == s.count
}

print(isSubsequence("abc", "ahbgdc"))
print(isSubsequence("axc", "ahbgdc"))